src/main/kotlin/com/sourcegraph/cody/config/SourcegraphInstanceLoginDialog.kt

```diff
@@ -0,0 +1,260 @@
+package com.sourcegraph.cody.config
+
+import com.intellij.collaboration.async.CompletableFutureUtil
+import com.intellij.collaboration.async.CompletableFutureUtil.errorOnEdt
+import com.intellij.collaboration.async.CompletableFutureUtil.submitIOTask
+import com.intellij.collaboration.async.CompletableFutureUtil.successOnEdt
+import com.intellij.openapi.application.ModalityState
+import com.intellij.openapi.application.invokeLater
+import com.intellij.openapi.components.service
+import com.intellij.openapi.observable.util.whenTextChanged
+import com.intellij.openapi.progress.EmptyProgressIndicator
+import com.intellij.openapi.progress.ProcessCanceledException
+import com.intellij.openapi.progress.ProgressIndicator
+import com.intellij.openapi.progress.ProgressManager
+import com.intellij.openapi.progress.util.ProgressIndicatorUtils
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.ui.DialogWrapper
+import com.intellij.openapi.ui.ValidationInfo
+import com.intellij.openapi.ui.setEmptyState
+import com.intellij.openapi.util.Disposer
+import com.intellij.ui.AnimatedIcon
+import com.intellij.ui.components.JBLabel
+import com.intellij.ui.components.JBTextField
+import com.intellij.ui.components.fields.ExtendableTextField
+import com.intellij.ui.dsl.builder.MAX_LINE_LENGTH_NO_WRAP
+import com.intellij.ui.dsl.builder.panel
+import com.intellij.ui.dsl.gridLayout.HorizontalAlign
+import com.intellij.ui.layout.not
+import com.intellij.ui.layout.selected
+import com.intellij.util.ui.UIUtil.getInactiveTextColor
+import com.sourcegraph.cody.api.SourcegraphApiRequestExecutor
+import com.sourcegraph.cody.api.SourcegraphAuthenticationException
+import com.sourcegraph.cody.auth.SourcegraphAuthService
+import com.sourcegraph.cody.auth.SsoAuthMethod
+import com.sourcegraph.cody.config.DialogValidationUtils.custom
+import com.sourcegraph.cody.config.DialogValidationUtils.notBlank
+import com.sourcegraph.common.AuthorizationUtil
+import com.sourcegraph.common.CodyBundle
+import com.sourcegraph.common.CodyBundle.fmt
+import java.awt.Component
+import java.awt.event.ActionEvent
+import java.net.UnknownHostException
+import java.util.concurrent.CompletableFuture
+import javax.swing.Action
+import javax.swing.JCheckBox
+
+class SourcegraphInstanceLoginDialog(
+    project: Project?,
+    parent: Component?,
+    private val defaultInstanceUrl: String = ""
```

Comment:
What is the purpose of this field? An empty string doesn't seem to be a reasonable default URL for anything.

Reply:
if the default is empty we show the empty text (e.g. `sourcegraph.<yourcompany>.com`)